### PR TITLE
Add -fhlsl-offsets, and -fresource-set-binding

### PR DESCRIPTION
**Pull request by dneto0**

### CODE REVIEW

File: glslc/src/main.cc [hunk 1]
```diff
@@ -336,6 +343,43 @@ int main(int argc, char** argv) {
                                               &arg_stage, &arg_base))
         return 1;
       set_binding_base(arg_stage, u_kind, arg_base);
+    } else if (arg == "-fresource-set-binding") {
+      auto need_three_args_err = [arg]() {
+        std::cerr << "glsc: error: Option " << arg
```
Comment (antiagainst) [inline glslc/src/main.cc]: We can actually embed the content of `arg` directly here to avoid the capture?

File: glslc/README.asciidoc [hunk 2]
```diff
@@ -218,6 +221,36 @@ If a stage is specified, only affects the specified stage.
 
 This only affects HLSL compilation.
 
+[[option-f-register-set-binding]]
```
Comment (dneto0) [inline glslc/README.asciidoc]: Fixed in next commit

Comment (dneto0) [inline glslc/README.asciidoc]: Fixed in next commit.

File: glslc/README.asciidoc [hunk 3]
```diff
@@ -218,6 +221,36 @@ If a stage is specified, only affects the specified stage.
...
+==== `-fresource-set-binding`
+
+Option `-fresource-set-binding [stage] <reg0> <set0> <binding1>` sets
```
Comment (dneto0) [inline glslc/README.asciidoc]: Good catch!  Fixed in next commit.

File: glslc/src/main.cc (hunk 1 above)
Comment (dneto0) [inline glslc/src/main.cc]: Fixed in next commit.

**Comment by dneto0** [comment]
All feedback addressed.  Also, rebased against master (including #351)

**Pull request merged**